master/internal/trial.go

```diff
@@ -871,14 +886,37 @@ func (t *trial) pushRendezvous(ctx *actor.Context) error {
 	return nil
 }
 
+func (t *trial) processContainerRunning(
+	ctx *actor.Context, msg sproto.TaskContainerStateChanged,
+) error {
+	ctx.Log().Infof("found container running: %s (rank %d)",
+		msg.Container.ID, len(t.containerRank))
+
+	t.containers[msg.Container.ID] = msg.Container
+	t.containerAddresses[msg.Container.ID] = msg.ContainerStarted.Addresses
+	t.containerRank[msg.Container.ID] = len(t.containerRank)
```

Comment:
blocking: we shouldn't be assigning rank based on the order in which containers connect. The chief container needs to match up with the container that is running process 0.

Reply:
I changed it to assign rank based on the index of `Allocation` and add the rank back to the name of pods.